### PR TITLE
moved Network(View)Client(Messages/Responses) to near_client

### PR DESCRIPTION
**Pull request by pompon0**
These actix messages are an implementation detail of near_client crate.

### CODE REVIEW

File: chain/network/src/client.rs [hunk 1]
```diff
@@ -19,400 +15,154 @@ use near_primitives::views::FinalExecutionOutcomeView;
...
-            Err(err) => {
-                tracing::error!("mailbox error: {err}");
-                Ok(())
-            }
-        }
+        accounts: Vec<(AnnounceAccount, Option<EpochId>)>,
+    ) -> Result<Vec<AnnounceAccount>, ReasonForBan>;
+}
+
+// Implementation of Client which doesn't do anything and never returns errors.
```
Comment (matklad) [inline chain/network/src/client.rs:85]:
```suggestion
/// Implementation of Client which doesn't do anything and never returns errors.
```

**Review by matklad — COMMENTED** [review]
What happened with `ReasonForBan`? 

Not to imply that anything is wrong with it, just that it looks like *something*'s changedwith our strategy of handling that, and it's a bit hard to understand what that was exactly from current PR description and diff )

File: chain/network/src/client.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/client.rs:85]: done

**Comment by pompon0** [comment]
ReasonForBan is returned only from a handful of handlers. I didn't know that until I implemented https://github.com/near/nearcore/pull/7895 . Interface in this PR is the final one. The next PR just reimplements Adapter.

File: chain/network/src/testonly/fake_client.rs [hunk 2]
```diff
@@ -22,96 +24,112 @@ pub enum Event {
...
-                let msg_type: &'static str = msg.into();
-                panic!("unsupported message {msg_type}")
-            }
-        }
+    async fn state_request_header(
+        &self,
+        _shard_id: ShardId,
+        _sync_hash: CryptoHash,
+    ) -> Result<Option<StateResponseInfo>, ReasonForBan> {
+        panic!("unimplemented");
```
Comment (matklad) [inline chain/network/src/testonly/fake_client.rs:51]: fwiw, there's `unimplemented!()`

File: chain/network/src/types.rs [hunk 3]
```diff
@@ -236,29 +230,15 @@ impl From<NetworkResponses> for PeerManagerMessageResponse {
...
-        hashes: Vec<CryptoHash>,
-        peer_id: PeerId,
-    },
+    BlockHeadersRequest { hashes: Vec<CryptoHash>, peer_id: PeerId },
     /// Request state header for given shard at given state root.
-    StateRequestHeader {
-        shard_id: ShardId,
-        sync_hash: CryptoHash,
-        target: AccountOrPeerIdOrHash,
-    },
```
Comment (matklad) [inline chain/network/src/types.rs:261]: Note that such formatting changes don't really help with reviewing gigantic PRs :)

As a general wish item, please try to find mode ways to minimize the diffs (not at the expense of landing awesome refactors obviously! ;) )

File: chain/network/src/types.rs [hunk 4]
```diff
@@ -267,23 +247,9 @@ pub enum NetworkRequests {
...
-        response: StateResponseInfo,
-    },
-    EpochSyncRequest {
-        peer_id: PeerId,
-        epoch_id: EpochId,
-    },
-    EpochSyncFinalizationRequest {
-        peer_id: PeerId,
-        epoch_id: EpochId,
-    },
```
Comment (matklad) [inline chain/network/src/types.rs:281]: Eg, here, killing epoch sync request could have been submitted as a separate PR which woudl be trivially approved. It's much faster to review a small non-trivial change, and a large trivail change, than a single diff which combines the two, requiring the user to separate "ok, so here we just s/foo/bar" from "ok, so here we are doing something special".

File: chain/network/src/types.rs [hunk 5]
```diff
@@ -435,71 +395,17 @@ pub enum NetworkResponses {
     RouteNotFound,
 }
 
-#[derive(actix::Message, Debug, strum::AsRefStr, strum::IntoStaticStr)]
-// TODO(#1313): Use Box
-#[allow(clippy::large_enum_variant)]
-#[rtype(result = "NetworkClientResponses")]
-pub enum NetworkClientMessages {
```
Comment (matklad) [inline chain/network/src/types.rs:442]: Similarly, not moving the home of this message would've reduced the diff substantially

File: chain/network/src/testonly/fake_client.rs (hunk 2 above)
Comment (pompon0) [reply chain/network/src/testonly/fake_client.rs:51]: done

File: chain/network/src/types.rs (hunk 5 above)
Comment (pompon0) [reply chain/network/src/types.rs:404]: done

File: chain/network/src/types.rs (hunk 3 above)
Comment (pompon0) [reply chain/network/src/types.rs]: ack

**Review by matklad — APPROVED** [review]
